scripts/suppression-budget.sh

```
@@ -0,0 +1,340 @@
+done
+
+# ---------- 逃生门：引用 scope 覆盖本门的 ADR → 豁免（绿）+ 入账 ----------
+EXEMPT=0; EXEMPT_ADR=""
+refs=$({ printf '%s\n' "${PR_TITLE:-}"
+         if [ -n "${PR_BODY_FILE:-}" ]; then cat "$PR_BODY_FILE" 2>/dev/null || true; fi; } \
+       | grep -oE '\bADR-[0-9]{4}\b' | sort -u || true)
+if [ -n "$refs" ]; then
+  if [ -n "${ADR_DIR:-}" ] && [ -d "$ADR_DIR" ]; then
+    for ref in $refs; do
+      num=${ref#ADR-}
+      for f in $(ls "$ADR_DIR" 2>/dev/null | grep -E "^ADR-${num}-" || true); do
```

Shellcheck (0.11.0)

[warning] Don't use ls | grep. Use a glob or a for loop with a condition to allow non-alphanumeric filenames.

(SC2010)